#!/usr/bin/env python
PACKAGE = "kinect_slam"
 
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("harris_window_size", int_t, 0, "harris_window_size", 5, 1, 15)
gen.add("harris_response_threshold", double_t, 0, "harris_response_threshold", 10, 0, 1000)
gen.add("harris_anms_flag", bool_t, 0, "harris_anms_flag", True);
gen.add("harris_anms_radius", int_t, 0, "harris_anms_radius", 26, 1, 50)
gen.add("harris_fix_number_flag", bool_t, 0, "harris_fix_number_flag", True);
gen.add("harris_number_of_points", int_t, 0, "harris_number_of_points", 3000, 1, 5000)
gen.add("harris_blur_size", int_t, 0, "harris_blur_size", 5, 1, 20)
gen.add("harris_blur_variance", int_t, 0, "harris_blur_variance", 4, 1, 20)

gen.add("descriptor_smooth_size", int_t, 0, "descriptor_smooth_size", 6, 1, 10)
gen.add("descriptor_brief_size", int_t, 0, "descriptor_brief_size", 1, 1, 3)
gen.add("descriptor_patch_size", int_t, 0, "descriptor_patch_size", 19, 15, 30)

exit(gen.generate(PACKAGE, "kinect_slam", "LandmarkExtractor"))
